.github/workflows/main.yml

```diff
@@ -273,3 +273,56 @@ jobs:
 
       - name: dart-format
         run: dart format -o none --set-exit-if-changed -l 110 .
+
+  # Create a new plain Rust project, add the SDK as single dependency and try to compile it.
+  # This tests whether the SDK compiles with the latest version of the dependencies that can be updated.
+  #
+  # Our checked-in Cargo.lock contains the specific combination of all direct and transitive dependency versions.
+  # This dependency tree snapshot was tested against during development and is what we release.
+  #
+  # However, when integrating the SDK in a new Rust project, Cargo not use our Cargo.lock. Instead, it will try to generate
+  # a new Cargo.lock based on our (and our dependencies') Cargo.toml files. This means that, where a dependency version range
+  # is used in a Cargo.toml, Cargo will try to upgrade it to the latest matching version. If this happens, this new dependency
+  # version may even result in the whole project failing to compile. In that case, the only solution is to manually pin
+  # the problematic dependencies to the last known good versions, in the application's Cargo.toml.
+  #
+  # Since this is the situation new projects are faced with when adding the SDK as a Rust dependency, we simulate it here
+  # to get an early warning signal, should any newer dependency cause it to fail to compile.
+  #
+  # See discussion at https://github.com/breez/breez-sdk/issues/969#issuecomment-2104700522
+  check-sdk-as-dependency:
```

Comment:
I think we can achieve that by adding `cargo install --path .` to the sdk-cli or I am missing something?

Reply:
Unfortunately not, because that would use the checked-in `Cargo.lock` of `sdk-cli`, so it wouldn't simulate the situation of "fresh new app with Breez SDK dependency".

AFAIK for applications, Rust relies on the available `Cargo.lock`. For libraries (like Breez SDK), it ignores it and reconstructs a fresh one from the library's `Cargo.toml`.